### ISSUE TITLE
Check out s2i; more straightforward image creation.

### ISSUE THREAD

**Issue by JamesKunstle** [sandiego-rh/explorer#132]
Asking contributors to know Docker as a prerequisite could inhibit their participation in the project. Check out the tool sourceToImage to see whether it would be advantageous to move to it. @codekow

**Comment by codekow** [comment]
@JamesKunstle Part of the benefit to s2i is sustainability. 

Instead of just managing your code you are maintaining containers. `Dockerfile` and `bash` tutorials are widely available. However, It doesn't teach best practices around containers and many devs just reuse what they find on stackoverflow without fully understanding the context.

I have to disagree with your assumption around s2i being potentially confusing and difficult for a user. It is opinionated and a paved path, but requires much less education for a developer than learning about container's (docker) ecosystem. s2i helps reduce a lot of onboarding education. 

Most devs do not live in a world of "we can rely on Docker and Bash scripts" - They usually only focus on their code - s2i is specifically designed for that. Maintaining `Dockerfile`s and `Bash` scripts is more work. You are increasing the amount of surface area you need to maintain.

s2i containers already provide a paved path for developers:
- Don't run as root
- Don't require any knowledge of containers
- Focus on running code, not maintaining containers

**Comment by JamesKunstle** [comment]
@codekow These are all very fair points. My opinion on s2i has changed pretty significantly in the last week because I finished DO180- I'm intrigued by it when before I was just confused by it. I see the value of the image being intelligently designed by a system rather than naively designed by a swe. 

My remaining hangup is the build-on-gh-actions-and-test process but I would be willing to bet that I can fit s2i into our existing process without a problem if that's the route we take.

Thank you for your thoughts and consideration!

**Comment by codekow** [comment]
@JamesKunstle Good news! :smile: `s2i` is just a command line tool. 

Using `s2i` in github actions should be straight forward. You can also use webhooks to trigger new builds in OCP with github. I'd be happy to show you how to do both in a working session.

### TL;DR
- https://github.com/openshift/source-to-image

**Comment by JamesKunstle** [comment]
probably not going to use s2i because we like compose
